actionview/lib/action_view/railtie.rb

```diff
@@ -48,6 +48,14 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "action_view.eager_load_templates" do |app|
+      ActiveSupport.on_load(:action_controller) do
+        if Rails.env.production? && app.config.action_view.eager_load_templates
```

Comment:
@egilburg When we got this feature to work on from @tenderlove, he said it should never be enabled in production. We could remove the hardcoded Rails.env.production?, but that would bring up some issues. The `config.eager_load` option needs to be on for our feature to work properly - if it is off (like it is by default in development), then the initializers won't be run on server startup, but on first request.

The options that we can think of are:
1. Leave it like this.
2. Remove the hardcode, and do nothing else.
3. Make the initializer check the value of `config.eager_load` and only run our code if it is on.

Which would be the best option?